### PR TITLE
Add dev requirement for `phpunit/phpunit`

### PR DESCRIPTION
**Pull request by phansys**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Comment by phansys** [comment]
The main goal is to have under control which version of PHPUnit is required to run the test suite.

File: circle.yml [hunk 1]
```diff
@@ -20,14 +21,14 @@ machine:
...
     - docker build -t gush .
 test:
   override:
-    - composer install
-    - wget https://phar.phpunit.de/phpunit.phar && chmod +x phpunit.phar
-    - php phpunit.phar --configuration phpunit.xml.dist --verbose
+    - composer install --prefer-dist --optimize-autoloader --no-interaction --no-scripts
+    - mkdir -p $CIRCLE_TEST_REPORTS/phpunit
+    - vendor/bin/phpunit --configuration phpunit.xml.dist --verbose --log-junit $CIRCLE_TEST_REPORTS/phpunit/junit.xml
     #- docker run -it -e GUSH_CONFIG=`cat ./.gush.yml | base64` gush -vvvv --help
```
Comment (cordoval) [inline circle.yml]: i would be impressed if you fix this line too 👍

File: composer.json [hunk 2]
```diff
@@ -26,7 +26,8 @@
     },
     "require-dev": {
         "mikey179/vfsStream": "~1.5.0",
-        "symfony/var-dumper": "~3.1.3"
+        "symfony/var-dumper": "~3.1.3",
+        "phpunit/phpunit": "^5.4"
```
Comment (cordoval) [inline composer.json]: we prefer here fixed numbers, use third digit stuff 5.4.2 or whatever is suit

File: circle.yml (hunk 1 above)
Comment (GrahamCampbell) [inline circle.yml]: :trollface:

**Comment by cordoval** [comment]
@phansys @GrahamCampbell is right, phpunit is known to create a whole lot of problems with global installs on composer, i don't care because it is now docker era, but still would be nice to avoid plugging a fixed phpunit, try rather to have a phar 💃 😊 or dockerize phpunit?

File: composer.json (hunk 2 above)
Comment (phansys) [inline composer.json]: Well @cordoval, I'll try to attend these comments today.

Comment (cordoval) [inline composer.json]: you rock @phansys

File: circle.yml (hunk 1 above)
Comment (phansys) [inline circle.yml]: @cordoval, fixed.

File: composer.json (hunk 2 above)
Comment (phansys) [inline composer.json]: You mean gush doesn't trust semver?

File: circle.yml (hunk 1 above)
Comment (phansys) [inline circle.yml]: I've configured my fork to run Circle CI tests in order to debug the `.gush.yml` path via SSH, but the issue regarding "docker: Error parsing reference: ..." is now solved.

Comment (phansys) [inline circle.yml]: Done.

File: circle.yml [hunk 3]
```diff
@@ -20,15 +21,15 @@ machine:
...
+    - "$HOME/docker"
   override:
     - docker build -t gush .
 test:
   override:
-    - composer install
-    - wget https://phar.phpunit.de/phpunit.phar && chmod +x phpunit.phar
-    - php phpunit.phar --verbose
-    #- docker run -it -e GUSH_CONFIG=`cat ./.gush.yml | base64` gush -vvvv --help
+    - composer install --prefer-dist --optimize-autoloader --no-interaction --no-scripts
```
Comment (cordoval) [inline circle.yml]: there is no need for --no-scripts because that is an ugly feature of composer and we don't use it

File: composer.json (hunk 2 above)
Comment (cordoval) [inline composer.json]: we do follow semver, we just don't fall for the dynamics and errors sometimes happen

Comment (phansys) [inline composer.json]: I understand you, but IMHO is too restrictive set a constraint against a specific version. Please, feel free to propose which version you'd like to lock.

**Comment by cordoval** [comment]
@GrahamCampbell though i agree with you, @phansys is doing some strides and if we think again, we should probably have a note or discourage global installations for the very reason that they are conflicting always, not just phpunit. That is what docker has wrought, independence from those problems.

File: composer.json (hunk 2 above)
Comment (cordoval) [inline composer.json]: I already merged it. This decision was taken years ago and nothing has changed since then in composer world.

**Pull request merged**